src/bitset.hpp

```diff
@@ -76,63 +93,144 @@ class Bitset {
 
+  // ** Subsplit Methods
   // These methods require the bitset to be a "subsplit bitset" of even length,
-  // consisting of two equal sized "chunks" representing the two sides of the
-  // subsplit.
+  // consisting of two equal sized "clades" representing the two sides of the
+  // subsplit. Clades are stored in a sorted order wrt to their lexigraphic taxon
+  // ordering: the smaller "left" or "sorted" clade stored in the 0th position, and the
+  // larger "right" or "rotated" clade in the 1st position.
   //
+  // Subsplit Constructor: each argument represents one of the clade of the subsplit.
+  // Each clade templated after Bitset constructor.
+  // Build a Subsplit bitset out of a compatible pair of clades.
+  static Bitset Subsplit(const Bitset &clade_0, const Bitset &clade_1);
+  // Builds Clades from string of "1" and "0" characters.
+  static Bitset Subsplit(const std::string clade_0, const std::string clade_1);
+  // Builds Clades of size `n` with only indices in the clade vectors set to true.
+  static Bitset Subsplit(const SizeVector clade_0, const SizeVector clade_1,
+                         const size_t n);
+  // Given two clades, create a subsplit with clades in sorted order by taxon
+  // representation.
+  static Bitset SubsplitOfOrderedClades(const Bitset &clade_0, const Bitset &clade_1);
+  // Comparator: Subsplits are sorting on the following:
+  // (1) The number of taxa in each of their subsplits.
+  // (2) The Bitset ordering of each of their respective unions.
+  // (3) The Bitset ordering of each or their sorted clades.
+  static int SubsplitCompare(const Bitset &subsplit_a, const Bitset &subsplit_b);
+  int SubsplitCompare(const Bitset &other) const;
   // Flip the order of the two sides of a subsplit.
-  Bitset RotateSubsplit() const;
-  size_t SubsplitChunkSize() const;
-  // Get the ith chunk of the subsplit.
-  Bitset SubsplitChunk(size_t i) const;
-  // Get "child 0" (the child bitset with the smaller binary representation)
-  // or "child 1" (the child bitset with the larger binary representation).
-  Bitset SubsplitGetChild(size_t i) const;
-  // Get a string of the bitset in the specified number of chunks, with each chunk
-  // separated by a "|".
-  std::string ToStringChunked(size_t chunk_count) const;
+  Bitset SubsplitRotate() const;
+  // Sorts clades of subsplit so that they are ordered by their taxon representation.
+  Bitset SubsplitSort() const;
+  // Gets the size of each of each clade. This is the same as the size of the whole
+  // taxon set.
+  size_t SubsplitGetCladeSize() const;
+  // Get clade according to its taxon ordering.
+  Bitset SubsplitGetClade(const size_t i) const;
+  // Get clade according to its binary ordering:
+  // Get "clade 0" (the clade bitset with the smaller binary representation)
+  // or "clade 1" (the clade bitset with the larger binary representation).
+  Bitset SubsplitGetOrderedClade(const size_t i) const;
```

Comment:
Yeah these two functions were originally `SubsplitChunk()` and `SubsplitGetChild()`.  Right now, it was written to return the ith clade according to its binary ordering for some reason.  

I think it would be best to have both `SubsplitGetClade` and `SubsplitGetOrderedClade` to return the same output for the same input, just that `SubsplitGetOrderedClade` would assert that the subsplit is indeed ordered, while `SubsplitGetClade` would allow for "rotated" subsplits.

Reply:
See above comment for response to this.